authority/provisioners.go

```diff
@@ -46,13 +47,43 @@ func (a *Authority) GetProvisioners(cursor string, limit int) (provisioner.List,
 func (a *Authority) LoadProvisionerByCertificate(crt *x509.Certificate) (provisioner.Interface, error) {
 	a.adminMutex.RLock()
 	defer a.adminMutex.RUnlock()
+	if p, err := a.unsafeLoadProvisionerFromDatabase(crt); err == nil {
+		return p, nil
+	}
+	return a.unsafeLoadProvisionerFromExtension(crt)
+}
+
+func (a *Authority) unsafeLoadProvisionerFromExtension(crt *x509.Certificate) (provisioner.Interface, error) {
 	p, ok := a.provisioners.LoadByCertificate(crt)
-	if !ok {
+	if !ok || p.GetType() == 0 {
 		return nil, admin.NewError(admin.ErrorNotFoundType, "unable to load provisioner from certificate")
 	}
 	return p, nil
 }
 
+func (a *Authority) unsafeLoadProvisionerFromDatabase(crt *x509.Certificate) (provisioner.Interface, error) {
+	// certificateDataGetter is an interface that can be use to retrieve the
+	// provisioner from a db or a linked ca.
+	type certificateDataGetter interface {
+		GetCertificateData(string) (*db.CertificateData, error)
+	}
```

Comment:
`certificateDataGetter` implies there could be more "data" related to certificates than just the provisioner metadata. Is that something to keep in mind for the future, e.g. would we extend the `CertificateData` to include other metadata if needed?

Reply:
I wasn't able to add more metadata to the certificate table because the structure is `key -> cert.Raw` so I created a new table that can contain more metadata, right now it only contains the provisioner used, but we can add more to it.